File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/inss/sobresalarios/TotalizadorOcorrenciaDeInssAtualizacao.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios;

import br.jus.trt8.pjecalc.base.comum.Utils;
import br.jus.trt8.pjecalc.negocio.comum.Total;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios.OcorrenciaDeInssAtualizacao;
import java.math.BigDecimal;
import java.util.Collection;

public class TotalizadorOcorrenciaDeInssAtualizacao {
    private Collection<? extends OcorrenciaDeInssAtualizacao> ocorrencias;
    private boolean isCalculado;
    private Total devidoCorrigido;
    private Total juros;
    private Total multa;
    private Total total;
    private Total pago;
    private Total devidoCorrigidoDiferenca;
    private Total jurosDiferenca;
    private Total multaDiferenca;
    private Total totalDiferenca;

    public TotalizadorOcorrenciaDeInssAtualizacao(Collection<? extends OcorrenciaDeInssAtualizacao> ocorrencias) {
        this.ocorrencias = ocorrencias;
        this.reset();
    }

    public void reset() {
        this.isCalculado = false;
    }

    private TotalizadorOcorrenciaDeInssAtualizacao calcular() {
        if (!this.isCalculado) {
            this.devidoCorrigido = Total.newInstance(true);
            this.juros = Total.newInstance(true);
            this.multa = Total.newInstance(true);
            this.total = Total.newInstance(true);
            this.pago = Total.newInstance(true);
            this.devidoCorrigidoDiferenca = Total.newInstance(true);
            this.jurosDiferenca = Total.newInstance(true);
            this.multaDiferenca = Total.newInstance(true);
            this.totalDiferenca = Total.newInstance(true);
            for (OcorrenciaDeInssAtualizacao ocorrenciaDeInssAtualizacao : this.ocorrencias) {
                if (Utils.naoNulo(ocorrenciaDeInssAtualizacao.getDevidoCorrigido())) {
                    this.devidoCorrigido.acumular(ocorrenciaDeInssAtualizacao.getDevidoCorrigido());
                }
                if (Utils.naoNulo(ocorrenciaDeInssAtualizacao.getJuros())) {
                    this.juros.acumular(ocorrenciaDeInssAtualizacao.getJuros());
                }
                if (Utils.naoNulo(ocorrenciaDeInssAtualizacao.getMulta())) {
                    this.multa.acumular(ocorrenciaDeInssAtualizacao.getMulta());
                }
                if (Utils.naoNulo(ocorrenciaDeInssAtualizacao.getTotal())) {
                    this.total.acumular(ocorrenciaDeInssAtualizacao.getTotal());
                }
                if (Utils.naoNulo(ocorrenciaDeInssAtualizacao.getPago())) {
                    this.pago.acumular(ocorrenciaDeInssAtualizacao.getPago());
                }
                if (Utils.naoNulo(ocorrenciaDeInssAtualizacao.getDevidoDiferenca())) {
                    this.devidoCorrigidoDiferenca.acumular(ocorrenciaDeInssAtualizacao.getDevidoDiferenca());
                }
                if (Utils.naoNulo(ocorrenciaDeInssAtualizacao.getJurosDiferenca())) {
                    this.jurosDiferenca.acumular(ocorrenciaDeInssAtualizacao.getJurosDiferenca());
                }
                if (Utils.naoNulo(ocorrenciaDeInssAtualizacao.getMultaDiferenca())) {
                    this.multaDiferenca.acumular(ocorrenciaDeInssAtualizacao.getMultaDiferenca());
                }
                if (!Utils.naoNulo(ocorrenciaDeInssAtualizacao.getTotalDiferenca())) continue;
                this.totalDiferenca.acumular(ocorrenciaDeInssAtualizacao.getTotalDiferenca());
            }
            this.isCalculado = true;
        }
        return this;
    }

    public BigDecimal getDevidoCorrigido() {
        return Utils.arredondarValorMonetario(this.calcular().devidoCorrigido.getValor());
    }

    public BigDecimal getJuros() {
        return Utils.arredondarValorMonetario(this.calcular().juros.getValor());
    }

    public BigDecimal getMulta() {
        return Utils.arredondarValorMonetario(this.calcular().multa.getValor());
    }

    public BigDecimal getTotal() {
        return Utils.arredondarValorMonetario(this.calcular().total.getValor());
    }

    public BigDecimal getPago() {
        return Utils.arredondarValorMonetario(this.calcular().pago.getValor());
    }

    public BigDecimal getDevidoCorrigidoDiferenca() {
        return Utils.arredondarValorMonetario(this.calcular().devidoCorrigidoDiferenca.getValor());
    }

    public BigDecimal getJurosDiferenca() {
        return Utils.arredondarValorMonetario(this.calcular().jurosDiferenca.getValor());
    }

    public BigDecimal getMultaDiferenca() {
        return Utils.arredondarValorMonetario(this.calcular().multaDiferenca.getValor());
    }

    public BigDecimal getTotalDiferenca() {
        return Utils.arredondarValorMonetario(this.calcular().totalDiferenca.getValor());
    }
}

File: src/lib/pjecalc/core/dominio/calculo/inss/sobresalarios/totalizador-ocorrencia-de-inss-atualizacao.ts
/**
 * PJe-Calc v2.15.1 — TotalizadorOcorrenciaDeInssAtualizacao
 * Porte 1:1 de: br.jus.trt8.pjecalc.negocio.dominio.calculo.inss.sobresalarios.TotalizadorOcorrenciaDeInssAtualizacao
 *
 * Ref Java: pjecalc-fonte/.../sobresalarios/TotalizadorOcorrenciaDeInssAtualizacao.java
 *
 * Acumula totais (devidoCorrigido, juros, multa, total, pago e as versões
 * "Diferenca") a partir de uma coleção de `OcorrenciaDeInssAtualizacao`.
 */
import Decimal from 'decimal.js';
import { arredondarValorMonetario } from '../../../../base/comum/utils';
import type { OcorrenciaDeInssAtualizacao } from './ocorrencia-de-inss-atualizacao';

const ZERO = new Decimal(0);

class Total {
  private valor: Decimal = ZERO;
  acumular(v: Decimal | null): void { if (v !== null) this.valor = this.valor.plus(v); }
  getValor(): Decimal { return this.valor; }
}

export class TotalizadorOcorrenciaDeInssAtualizacao {
  private ocorrencias: Iterable<OcorrenciaDeInssAtualizacao>;
  private isCalculado = false;
  private devidoCorrigido = new Total();
  private juros = new Total();
  private multa = new Total();
  private total = new Total();
  private pago = new Total();
  private devidoCorrigidoDiferenca = new Total();
  private jurosDiferenca = new Total();
  private multaDiferenca = new Total();
  private totalDiferenca = new Total();

  constructor(ocorrencias: Iterable<OcorrenciaDeInssAtualizacao>) {
    this.ocorrencias = ocorrencias;
  }

  reset(): void { this.isCalculado = false; }

  private calcular(): this {
    if (this.isCalculado) return this;
    this.devidoCorrigido = new Total();
    this.juros = new Total();
    this.multa = new Total();
    this.total = new Total();
    this.pago = new Total();
    this.devidoCorrigidoDiferenca = new Total();
    this.jurosDiferenca = new Total();
    this.multaDiferenca = new Total();
    this.totalDiferenca = new Total();
    for (const oc of this.ocorrencias) {
      this.devidoCorrigido.acumular(oc.getDevidoCorrigido());
      this.juros.acumular(oc.getJuros());
      this.multa.acumular(oc.getMulta());
      this.total.acumular(oc.getTotal());
      this.pago.acumular(oc.getPago());
      this.devidoCorrigidoDiferenca.acumular(oc.getDevidoDiferenca());
      this.jurosDiferenca.acumular(oc.getJurosDiferenca());
      this.multaDiferenca.acumular(oc.getMultaDiferenca());
      this.totalDiferenca.acumular(oc.getTotalDiferenca());
    }
    this.isCalculado = true;
    return this;
  }

  getDevidoCorrigido(): Decimal { return arredondarValorMonetario(this.calcular().devidoCorrigido.getValor()); }
  getJuros(): Decimal { return arredondarValorMonetario(this.calcular().juros.getValor()); }
  getMulta(): Decimal { return arredondarValorMonetario(this.calcular().multa.getValor()); }
  getTotal(): Decimal { return arredondarValorMonetario(this.calcular().total.getValor()); }
  getPago(): Decimal { return arredondarValorMonetario(this.calcular().pago.getValor()); }
  getDevidoCorrigidoDiferenca(): Decimal { return arredondarValorMonetario(this.calcular().devidoCorrigidoDiferenca.getValor()); }
  getJurosDiferenca(): Decimal { return arredondarValorMonetario(this.calcular().jurosDiferenca.getValor()); }
  getMultaDiferenca(): Decimal { return arredondarValorMonetario(this.calcular().multaDiferenca.getValor()); }
  getTotalDiferenca(): Decimal { return arredondarValorMonetario(this.calcular().totalDiferenca.getValor()); }
}
